src/client/Client.cc

```diff
@@ -2878,7 +2881,7 @@ void Client::_handle_full_flag(int64_t pool)
 
 void Client::handle_osd_map(const MConstRef<MOSDMap>& m)
 {
-  std::scoped_lock cl(client_lock);
+  std::unique_lock cl(client_lock);
```

Comment:
Is this needed since the context callback is being queued via the finisher?

Reply:
Yeh, I mean it can hold the client lock while calling op_cancel_writes() since _write() wraps the context callback via Finisher.

Reply:
I tested this locally and it works. So I have removed the commit which calls the op_cance_writes without taking client_lock

Reply:
op, do you mean without releasing `client_lock` (handle_osd_map() does acquire `client_lock`).

Reply:
No I mean, I have removed the commit which was calling op_cancel_writes by unlocking the `client_lock` which caused the lock order dependency.

Reply:
Checked your latest push. We are talking about the same thing, so 👍